### ISSUE TITLE
Style/MixinGrouping warns for method calls which share the same name with the built-in mixin methods

### ISSUE THREAD

**Issue by attilahorvath** [bbatsov/rubocop#4206]
The `Style/MixinGrouping` cop doesn't take into consideration that the mixin methods (`include`, `extend` and `prepend`) might be used for different purposes in custom classes and erroneously warns about their usage.

Such Ruby methods might seem like code smells but they are already part of some DSLs, e.g. the RSpec config: http://www.rubydoc.info/github/rspec/rspec-core/RSpec/Core/Configuration#include-instance_method

Here is a toy example to illustrate the problem:

```ruby
class C
  def initialize
    include
    extend
    prepend
  end

  def include; end

  def extend; end

  def prepend; end
end
```

--------

## Expected behavior

RuboCop should not raise any offenses when run against the example.

## Actual behavior

RuboCop reports the following offenses:

```
Inspecting 1 file
C

Offenses:

test.rb:3:5: C: Put include mixins in separate statements.
    include
    ^^^^^^^
test.rb:4:5: C: Put extend mixins in separate statements.
    extend
    ^^^^^^
test.rb:5:5: C: Put prepend mixins in separate statements.
    prepend
    ^^^^^^^
```

## Steps to reproduce the problem

Run the latest RuboCop with default configuration against the example provided.

## RuboCop version

```
$ rubocop -V
0.48.0 (using Parser 2.4.0.0, running on ruby 2.3.1 x86_64-linux)
```

**Comment by attilahorvath** [comment]
Edited the issue because I realised that the built-in mixin methods are not even overridden in this case - they are class methods so they would still be intact. I was just defining instance methods which happened to share the same names.

I think this behaviour could be fixed by only checking for these method calls in the class bodies but not in the method bodies.

**Comment by andreaswachowski** [comment]
This looks like it has already been fixed with PR #4177.

**Comment by attilahorvath** [comment]
Strange, I searched for this issue but didn't see that PR. Closing this.
